manifest/kcp-front-proxy.yaml

```diff
@@ -0,0 +1,258 @@
+  privateKey:
+    algorithm: RSA
+    size: 2048
+  issuerRef:
+    name: kcp-pki-bootstrap
+    kind: Issuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: kcp-client-issuer
+spec:
+  ca:
+    secretName: kcp-client-ca
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: kcp-front-proxy-kcp-client-cert
+spec:
+  secretName: kcp-front-proxy-kcp-client-cert
+  duration: 2160h0m0s # 90d
+  renewBefore: 360h0m0s # 15d
+  subject:
+    organizations:
+      - redhat
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - client auth
+  dnsNames:
+    - "kcp-front-proxy"
+  issuerRef:
+    name: kcp-requestheader-client-issuer
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: kcp-front-proxy-virtual-workspaces-client-cert
+spec:
+  secretName: kcp-front-proxy-virtual-workspaces-client-cert
+  duration: 2160h0m0s # 90d
+  renewBefore: 360h0m0s # 15d
+  subject:
+    organizations:
+      - redhat
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - client auth
+  dnsNames:
+    - "kcp-front-proxy"
+  issuerRef:
+    name: kcp-requestheader-client-issuer
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kcp-front-proxy
+spec:
+  ports:
+    - protocol: TCP
+      name: kcp-front-proxy
+      port: 8443
+      targetPort: 8443
+  selector:
+    app: kcp-front-proxy
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kcp-front-proxy-config
+data:
+  path-mapping.yaml: |
+    - path: /services/
+      backend: https://kcp:6444
+      backend_server_ca: /etc/virtual-workspaces/tls/ca.crt
+      proxy_client_cert: /etc/kcp-front-proxy/requestheader-client/tls/virtual-workspaces/tls.crt
+      proxy_client_key: /etc/kcp-front-proxy/requestheader-client/tls/virtual-workspaces/tls.key
+    - path: /
+      backend: https://kcp:6443
+      backend_server_ca: /etc/kcp/tls/ca.crt
+      proxy_client_cert: /etc/kcp-front-proxy/requestheader-client/tls/kcp/tls.crt
+      proxy_client_key: /etc/kcp-front-proxy/requestheader-client/tls/kcp/tls.key
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kcp-front-proxy
+  labels:
+    app: kcp-front-proxy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kcp-front-proxy
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: kcp-front-proxy
+    spec:
+      containers:
+      - name: kcp-front-proxy
+        image: ghcr.io/kcp-dev/kcp:latest
+        ports:
+        - containerPort: 8443
+        command:
+        - /kcp-front-proxy
+        args:
+        - --secure-port=8443
+        - --tls-private-key-file=/etc/kcp-front-proxy/tls/tls.key
+        - --tls-cert-file=/etc/kcp-front-proxy/tls/tls.crt
+        - --client-ca-file=/etc/kcp-front-proxy/client/tls/ca.crt
+        - --mapping-file=/etc/kcp-front-proxy/config/path-mapping.yaml
+        - --v=4
```

Comment:
Let's change this to 6 so the proxy will log basic info about requests to the backends.